### DISCUSSION TITLE
@parcel/transformer-js : Inlining fs.promises.readFile() ?

### DISCUSSION THREAD

**Discussion by jy95** [parcel-bundler/parcel#8679] [Q&A]
Hello,

I have a Typescript situation with such piece of code : 

```ts
// src/private/moduleA.ts
import { readFile } from "fs/promises";
import path from "path";

export async function readSchema() {
     let text = await readFile(path.join(__dirname, "..", "mySchema.json"), "utf-8");
     let json = JSON.parse(text.toString());
     return json;
}

// src/moduleB.ts
import { readSchema } from "./private/moduleA";

// Some code
let schema = await readSchema();

// src/moduleC.ts
import { readSchema } from "./private/moduleA";

// Some code
let schema = await readSchema();

```

I would like to have something similar to [Inlining fs.readFileSync](https://parceljs.org/features/node-emulation/#inlining-fs.readfilesync) and switching to fs.readFileSync is not an option at all. readSchema is also a function used by other modules by the way

Any ideas ?

Thanks in advance

**Comment by mischnic** [comment]
> switching to fs.readFileSync is not a option 

Why is that not an option?

Because after running Parcel on a version with `readFilesync´, the code looks like this
```js
export function readSchema() {
     let text = "some long string";
     let json = JSON.parse(text.toString());
     return json;
}
```

**Reply by jy95** [reply]
Well, in general, programmers are warned against sync stuff : 

- [Why sync execution is bad](https://advancedweb.hu/do-not-use-fs-sync-methods-in-javascript-use-fs-promises-instead/)
- [The 5 Benefits of Asynchronous Learning](https://curve.mit.edu/5-benefits-asynchronous-learning#:~:text=An%20asynchronous%20course%20provides%20the,in%20a%20synchronous%20online%20course.)
- ...

I found a bit weird that Parcel.js don't inline the async version

**Reply by mischnic** [reply]
That's what I tried to say with my example: it's not actually a synchronous FS operation after inlining.

There's no real reason why Parcel doesn't inline the async version, there was just no need so far.
